webapp/components/survey/TaxonomyDetails/store/actions/useUpdate.js

```diff
@@ -15,17 +16,15 @@ export const useUpdate = ({ setState }) => {
 
   return useCallback(({ key, value, state }) => {
     const taxonomy = State.getTaxonomy(state)
+    const taxonomyUuid = Taxonomy.getUuid(taxonomy)
 
     const taxonomyUpdated = State.assocTaxonomyProp({ key, value })(taxonomy)
     setState(State.assocTaxonomy(taxonomyUpdated)(state))
 
     const action = async () => {
-      await axios.put(`/api/survey/${surveyId}/taxonomies/${Taxonomy.getUuid(taxonomy)}`, {
-        key,
-        value,
-      })
+      await API.updateTaxonomy({ surveyId, taxonomyUuid, data: { key, value } })
     }
     dispatch(SurveyActions.metaUpdated())
-    dispatch(debounceAction(action, Taxonomy.getUuid(taxonomy)))
+    dispatch(debounceAction(action, taxonomyUuid))
```

Comment:
should we use a more complex key when debounceAction is called, just to avoid collisions? something like `taxonomy_update_${taxonomyUuid}`?

Reply:
i dindt know, sure i will change :)